pages/api/mentors/[mentor_id].js

```diff
@@ -1,6 +1,14 @@
 // GET: /pages/api/mentors/[mentor_id].js
+import { db } from '../../../utils/api/firebase.config';
+import { collection, doc, getDoc } from "firebase/firestore";
 
-export default function getMentorInfo(req, res) {
+export default async function getMentorInfo(req, res) {
   const { mentor_id } = req.query;
-  // open to add query
+  try {
+    const docRef = doc(db, 'mentors', mentor_id);
+    const docSnap = await getDoc(docRef);
+    res.status(200).json(docSnap.data());
+  } catch (err) {
+    res.status(400).send(`Error retrieving user info: ${err}`);
```

Comment:
Nit: more accurate error codes. Not sure how firebase works with empty queries but can also add a 404 error response for empty queries. No need to edit now but something to keep in mind for later